### PR TITLE
--save-not-found option

### PR DESCRIPTION
**Pull request by nmaier**
Recently I found myself in the situation where I had session files which contained a lot of "expired" files (which are 404). Keeping those in the session is not great, especially if you restart aria2 and it then of course would try to download those 404 files again, putting strain on the server and making things slower. Manually tracking and removing the items from the session files is extremely cumbersome, and since I run that aria2 without RPC, I cannot properly script it either.

Therefore, I propose this `--save-not-found` option, which I left at `true` by default, which when disabled (`false`) will skip any not found files when serializing a session.

@tatsuhiro-t any comments or objections?

### CODE REVIEW

File: src/SessionSerializer.cc [hunk 1]
```diff
@@ -277,38 +277,32 @@ bool SessionSerializer::save(IOFile& fp) const
...
+      case error_code::FINISHED:
+      case error_code::REMOVED:
+        save = dr->option->getAsBool(PREF_FORCE_SAVE);
+        break;
+      case error_code::IN_PROGRESS:
+        save = saveInProgress_;
+        break;
+      case error_code::RESOURCE_NOT_FOUND:
+      case error_code::MAX_FILE_NOT_FOUND:
+        save = dr->option->getAsBool(PREF_SAVE_NOT_FOUND);
```
Comment (tatsuhiro-t) [inline src/SessionSerializer.cc]: For consistency, it should take into account saveError_.

**Comment by tatsuhiro-t** [comment]
Green light.  No objection from me.

**Comment by nmaier** [comment]
I'll make the `saveError_` change and commit

**Comment by nmaier** [comment]
https://github.com/aria2/aria2/commit/414dd14decff7429ce94f18721f4c1afe607b6ea

**Pull request closed without merging**